walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,75 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    int duration = (int) request.getDuration();
+    Set<String> attendees = new HashSet<String>(request.getAttendees());
+    List<TimeRange> initalRanges = new ArrayList<TimeRange>();
+    List<TimeRange> results = new ArrayList<TimeRange>();
+    
+    for(int start = 0; start <= TimeRange.END_OF_DAY; start += duration) {
```

Comment:
Consider the following: If you request a 60 minute time slot, this code would only check:

0-60
60-120
120-180,

etc

note that you are missing potential valid solutions:

0-60
30-90
60-120

etc

Reply:
After looking at the events starter file is seems that events are required to have their duration be  a factor of 30 so I changed the ranges in initalRanges to have a duration of 30 and wrote a test